config/application.rb

```diff
@@ -18,6 +18,18 @@
 # you've limited to :test, :development, or :production.
 Bundler.require(*Rails.groups)
 
+ActiveRecord::AssociatedObject.class_eval {
+  def self.generates_token(expires_in:, &)
+    purpose = attribute_name
+    record.generates_token_for(purpose, expires_in:, &)
+
+    define_singleton_method(:find_by_token)  { find_by_token_for(purpose, _1) }
+    define_singleton_method(:find_by_token!) { find_by_token_for!(purpose, _1) }
+
+    define_method(:token) { record.generate_token_for(purpose) }
+  end
+}
```

Comment:
I can't quite figure out if this whole thing is going too far, so I'm curious to hear your thoughts.

I've also been thinking about whether this should just be `signed_id` delegation instead of tokens.

Reply:
Seems like you're in!